### PR TITLE
chore: Add filter and sorting utilities

### PR DESCRIPTION
**Pull request by FernandoAranda**
Issue #, if available:
GALI-1841
Description of changes:
Add sorting and filter utilities for list services

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/base/src/utilities/queryParamsFiltersUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import {
+  getFilterQueryParams,
+  getSortQueryParams,
+  validateSingleSortAndFilter
+} from './queryParamsFiltersUtils';
+
+describe('Query params filters utils', () => {
+  describe('validateSingleSortAndFilter', () => {
+    test('should throw exception when request has filter and sorting', () => {
+      expect(() => validateSingleSortAndFilter({ name: { eq: 'Sauron' } }, { status: 'desc' })).toThrow(
```
Comment (maghirardelli) [inline workbench-core/base/src/utilities/queryParamsFiltersUtils.test.ts:14]: interesting test value choice...

File: workbench-core/base/src/utilities/queryParamsFiltersUtils.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { marshall } from '@aws-sdk/util-dynamodb';
+import Boom from '@hapi/boom';
+import { QueryParams } from '../aws/helpers/dynamoDB/dynamoDBService';
```
Comment (maghirardelli) [inline workbench-core/base/src/utilities/queryParamsFiltersUtils.ts:7]: Also I think `queryParams.ts` should go into your interfaces folder. I had it in constants but interfaces makes more sense. Can you move when you merge `develop` into this branch?

File: workbench-core/base/src/interfaces/FilterRequest.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { QueryParameterFilter } from './queryParameterFilter';
+
+export interface FilterRequest {
```
Comment (maghirardelli) [inline workbench-core/base/src/interfaces/FilterRequest.ts:8]: nit: file name should be `filterRequest.ts`

File: workbench-core/base/src/interfaces/SortRequest.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+export interface SortRequest {
```
Comment (maghirardelli) [inline workbench-core/base/src/interfaces/SortRequest.ts:6]: nit: file name should be `sortRequest.ts`

File: workbench-core/base/src/interfaces/queryParameterFilter.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+/************
+ *
+ * Only one operator can be defined by property, if multiple operators are defined, dynamo service will throw an exception.
+ *
+ ************/
+export interface QueryParameterFilter<T> {
```
Comment (maghirardelli) [inline workbench-core/base/src/interfaces/queryParameterFilter.ts:11]: Why do you use the generic type here? Won't these values always be strings?

File: workbench-core/base/src/utilities/queryParamsFiltersUtils.ts (hunk 2 above)
Comment (FernandoAranda) [reply workbench-core/base/src/utilities/queryParamsFiltersUtils.ts]: added to interfaces

File: workbench-core/base/src/interfaces/FilterRequest.ts (hunk 3 above)
Comment (FernandoAranda) [reply workbench-core/base/src/interfaces/FilterRequest.ts:8]: fixed in commit

File: workbench-core/base/src/interfaces/SortRequest.ts (hunk 4 above)
Comment (FernandoAranda) [reply workbench-core/base/src/interfaces/SortRequest.ts:6]: fixed in commit

File: workbench-core/base/src/interfaces/FilterRequest.ts (hunk 3 above)
Comment (maghirardelli) [reply workbench-core/base/src/interfaces/FilterRequest.ts:8]: I see the filenames in import statements are fixed, but are the files themselves now starting with lowercase? they still look like uppercase in the Files tab here.

Comment (maghirardelli) [reply workbench-core/base/src/interfaces/FilterRequest.ts:8]: ^^ I don't think you actually changed the file names which is why the build and test GH action failed btw.

File: workbench-core/base/src/interfaces/filterRequest.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { QueryParameterFilter } from './queryParameterFilter';
+
+export interface FilterRequest {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (bhudnell) [inline workbench-core/base/src/interfaces/filterRequest.ts:9]: dont need this line anymore

**Review by bhudnell — APPROVED** [review]

**Review by bhudnell — APPROVED** [review]

**Review by maghirardelli — APPROVED** [review]
LGTM!

**Pull request merged**